org-jira.el

```diff
@@ -859,57 +908,60 @@ Expects input in format such as: [2017-04-05 Wed 01:00]--[2017-04-05 Wed 01:46]
         ))))
 
 ;;;###autoload
-(defun org-jira-update-worklogs-from-org-clocks ()
-  "Update or add a worklog based on the org clocks."
-  (interactive)
-  (let ((issue-id (org-jira-get-from-org 'issue 'key)))
-    (ensure-on-issue-id
-     issue-id
-     (search-forward (format ":%s:" (or (org-clock-drawer-name) "LOGBOOK"))  nil 1 1)
-     (org-beginning-of-line)
-     (org-cycle 1)
-     (while (search-forward "CLOCK: " nil 1 1)
-       (let ((org-time (buffer-substring-no-properties (point) (point-at-eol))))
-         (forward-line)
-         ;; See where the stuff ends (what point)
-         (let (next-clock-point)
-           (save-excursion
-             (search-forward-regexp "\\(CLOCK\\|:END\\):" nil 1 1)
-             (setq next-clock-point (point)))
-           (let ((clock-content
-                  (buffer-substring-no-properties (point) next-clock-point)))
-
-             ;; @todo :optim: This is inefficient, calling the resync on each update/insert event,
-             ;; ideally we would track and only insert/update changed entries, as well
-             ;; only call a resync once (when the entire list is processed, which will
-             ;; basically require a dry run to see how many items we should be updating.
-
-             ;; Update via jiralib call
-             (let* ((worklog (org-jira-org-clock-to-jira-worklog org-time clock-content))
-                    (comment-text (cdr (assoc 'comment worklog)))
-                    (comment-text (if (string= (org-trim comment-text) "") nil comment-text)))
-               (if (cdr (assoc 'worklog-id worklog))
-                   (jiralib-update-worklog
-                    issue-id
-                    (cdr (assoc 'worklog-id worklog))
-                    (cdr (assoc 'started worklog))
-                    (cdr (assoc 'time-spent-seconds worklog))
-                    comment-text
-                    (cl-function
-                     (lambda (&rest data &allow-other-keys)
-                       (org-jira-update-worklogs-for-current-issue))))
-                 ;; else
-                 (jiralib-add-worklog
-                  issue-id
-                  (cdr (assoc 'started worklog))
-                  (cdr (assoc 'time-spent-seconds worklog))
-                  comment-text
-                  (cl-function
-                   (lambda (&rest data &allow-other-keys)
-                     (org-jira-update-worklogs-for-current-issue))))
-                 )
-               )))))
-     )))
+(defun org-jira-update-worklogs-from-org-clocks (&optional force)
+  "Update or add a worklog based on the org clocks when `org-jira-worklog-sync-p' is not nil.
+If you have dosabled the worklog sync globally, but you would still like to triger it 
```

Comment:
Few typos in comment - I'm also not sure if this is the correct approach (to have a per-function override for a state controlled behavior).

This function's existing purpose is to perform the indicated behavior of updating worklogs from org clocks when called (imagine a situation where the user does not 'force' it and has the sync set to nil - they can now call a function that does absolutely nothing).

Controlling it at the invocation level makes more sense to me (a chained action xyz will not occur unless the predicates for it have been met).